### ISSUE TITLE
`PMIx_jll` 4.2.7 seems to be broken (or something has changed)

### ISSUE THREAD

**Issue by Sbozzolo** [JuliaPackaging/Yggdrasil#7605]
On two remote clusters I have access to, `PMIx_jll` 4.2.7 cannot be loaded with the following error:
```
ERROR: InitError: could not load library "/home/gbozzola/.julia/artifacts/b64ad2ff5bdc8b41a59009d2b5547838cfde4f5d/lib/libpmix.so"
libevent_core-2.1.so.7: cannot open shared object file: No such file or directory
Stacktrace:
  [1] dlopen(s::String, flags::UInt32; throw_error::Bool)
    @ Base.Libc.Libdl ./libdl.jl:117
  [2] dlopen(s::String, flags::UInt32)
    @ Base.Libc.Libdl ./libdl.jl:116
  [3] macro expansion
    @ ~/.julia/packages/JLLWrappers/pG9bm/src/products/library_generators.jl:63 [inlined]
  [4] __init__()
    @ PMIx_jll ~/.julia/packages/PMIx_jll/UrLDH/src/wrappers/x86_64-linux-gnu.jl:12
  [5] register_restored_modules(sv::Core.SimpleVector, pkg::Base.PkgId, path::String)
    @ Base ./loading.jl:1115
  [6] _include_from_serialized(pkg::Base.PkgId, path::String, ocachepath::String, depmods::Vector{Any})
    @ Base ./loading.jl:1061
  [7] _require_search_from_serialized(pkg::Base.PkgId, sourcepath::String, build_id::UInt128)
    @ Base ./loading.jl:1506
  [8] _require(pkg::Base.PkgId, env::String)
    @ Base ./loading.jl:1783
  [9] _require_prelocked(uuidkey::Base.PkgId, env::String)
    @ Base ./loading.jl:1660
 [10] macro expansion
    @ ./loading.jl:1648 [inlined]
 [11] macro expansion
    @ ./lock.jl:267 [inlined]
 [12] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:1611
during initialization of module PMIx_jll
```
To be noted that
```
$ find .julia -name "*libevent_core*"
.julia/artifacts/e9feeabfa989549ac1dac61e4df5f2849149ad6d/lib/libevent_core-2.1.so.7
.julia/artifacts/e9feeabfa989549ac1dac61e4df5f2849149ad6d/lib/libevent_core-2.1.so.7.0.1
.julia/artifacts/e9feeabfa989549ac1dac61e4df5f2849149ad6d/lib/libevent_core.a
.julia/artifacts/e9feeabfa989549ac1dac61e4df5f2849149ad6d/lib/libevent_core.so
.julia/artifacts/e9feeabfa989549ac1dac61e4df5f2849149ad6d/lib/pkgconfig/libevent_core.pc
```
I am using Julia 1.9.3 on both systems. `PMIx_jll` works on my laptop, but I have `libevent_core-2.1.so.7` in my `/lib64`.

Pinning to `4.1.0` works.

If `PMIx_jll` is not broken upstream, I am guessing something has changed that breaks our systems. I tried to look at the repo, but I don't understand it enough to see what has changed. Help in this direction would be great!

Thanks in advance.

**Comment by giordano** [comment]
More information:
```
julia> using PMIx_jll
ERROR: InitError: could not load library "/home/cceamgi/.julia/artifacts/b64ad2ff5bdc8b41a59009d2b5547838cfde4f5d/lib/libpmix.so"
libevent_core-2.1.so.7: cannot open shared object file: No such file or directory
[...]

julia> using DependencyWalker

julia> Library("/home/cceamgi/.julia/artifacts/b64ad2ff5bdc8b41a59009d2b5547838cfde4f5d/lib/libpmix.so")
◼ /home/cceamgi/.julia/artifacts/b64ad2ff5bdc8b41a59009d2b5547838cfde4f5d/lib/libpmix.so
  ◼ /home/cceamgi/.julia/juliaup/julia-1.9.3+0.x64.linux.gnu/bin/../lib/julia/libopenlibm.so
    ◼ /lib64/libc.so.6
      ◼ /lib64/ld-linux-x86-64.so.2
  ◼ /home/cceamgi/.julia/artifacts/641f1e03ca8b7723a67832866a3bb958bc4e0555/lib/libhwloc.so
    ◼ /home/cceamgi/.julia/juliaup/julia-1.9.3+0.x64.linux.gnu/bin/../lib/julia/libopenlibm.so
      ◼ /lib64/libc.so.6
        ◼ /lib64/ld-linux-x86-64.so.2
    ◼ /lib64/libc.so.6
      ◼ /lib64/ld-linux-x86-64.so.2
  ◼ /lib64/libpthread.so.0
    ◼ /lib64/ld-linux-x86-64.so.2
    ◼ /lib64/libc.so.6
      ◼ /lib64/ld-linux-x86-64.so.2
  ◼ /lib64/librt.so.1
    ◼ /lib64/libpthread.so.0
      ◼ /lib64/ld-linux-x86-64.so.2
      ◼ /lib64/libc.so.6
        ◼ /lib64/ld-linux-x86-64.so.2
    ◼ /lib64/libc.so.6
      ◼ /lib64/ld-linux-x86-64.so.2
  ✗ libutil.so (NOT FOUND)
  ◼ /lib64/libdl.so.2
    ◼ /lib64/ld-linux-x86-64.so.2
    ◼ /lib64/libc.so.6
      ◼ /lib64/ld-linux-x86-64.so.2
  ✗ libevent_core-2.1.so (NOT FOUND)
  ◼ /lib64/libc.so.6
    ◼ /lib64/ld-linux-x86-64.so.2
  ✗ libevent_pthreads-2.1.so (NOT FOUND)
```

**Comment by giordano** [comment]
There's something very wrong in either pmix or libevent: there's no `libevent_core-2.1.so` nor `libevent_pthreads-2.1.so` in libevent package.

**Comment by giordano** [comment]
Fixed by #7609.
